src/include/txn_inline.h

```diff
@@ -295,14 +295,32 @@ __wt_txn_op_delete_apply_prepare_state(WT_SESSION_IMPL *session, WT_REF *ref, bo
     WT_REF_UNLOCK(ref, previous_state);
 }
 
+/*
+ * __wt_txn_op_delete_commit_apply_page_del_timestamp --
+ *     Apply the correct start and durable timestamps to the page delete structure.
+ */
+static inline void
+__wt_txn_op_delete_commit_apply_page_del_timestamp(WT_SESSION_IMPL *session, WT_REF *ref)
+{
+    WT_PAGE_DELETED *page_del;
+    WT_TXN *txn;
+
+    txn = session->txn;
+    page_del = ref->page_del;
```

Comment:
Could you add an assertion in here about the REF state? I think it should be locked? Given that different callers now have different ways of locking, it feels worth recording the contract via an assertion.

Reply:
Added here
https://github.com/wiredtiger/wiredtiger/blob/ff2d37ea8db649118482cc0b7086f092b8fe72f6/src/include/txn_inline.h#L311